ui/src/main/java/org/pentaho/di/ui/spoon/tree/TreeManager.java

```diff
@@ -188,6 +190,11 @@ private <T> void render( TreeNode treeNode, T tree ) {
     TreeItem childTreeItem = createTreeItem( treeNode, tree );
     if ( treeNode.hasChildren() ) {
       for ( TreeNode childTreeNode : treeNode.getChildren() ) {
+        Repository repository = Spoon.getInstance().getRepository();
+        if( childTreeNode.getLabel().equals( "VFS Connections" ) && repository!=null
```

Comment:
Generally, it seems that this is being done after the fact: that we should be trying to _prevent_ whoever is adding the VFS Connections node to the tree from doing so, instead of "harcoding" logic to _remove_ a specific tree node in this central location. AFAICT, up until now, the tree manager has no specific knowledge of VFS Connections or other specific nodes.

Otherwise, if this is the only possible location to intervene, see the following smaller issues:
1. Check for `repository != null` first.
2. Check the tree node using some identifier, and not the label, given that will be locale dependent.
3. Formatting is off.

Reply:
#2 i made similar observation with this comment - https://github.com/pentaho/pentaho-kettle/pull/9219#discussion_r1517976933.

But that might be irrelevant if TreeNode "VFS Connections" is never added to `rootNodes` as I commented in https://github.com/pentaho/pentaho-kettle/pull/9219#discussion_r1517956769

Reply:
Changed the location of code, please re-review